File: app/src/main/java/com/lbynet/Phokus/utils/Timer.java
package com.lbynet.Phokus.utils;

public class Timer {

    private long startTime = 0;
    private long endTime = 0;
    private boolean isBusy = false;
    private String name;

    public Timer(String name) {
        this.name = name;
        startTime = System.currentTimeMillis();
        isBusy = true;
    }

    public Timer() {
        this("DEFAULT TIMER");
    }

    public void start() {
        startTime = System.currentTimeMillis();
        isBusy = true;
    }

    public boolean isBusy() {
        return isBusy;
    }

    public void stop() {
        endTime = System.currentTimeMillis();
        isBusy = false;
    }

    public long getElaspedTimeInMs() {

        if(isBusy) endTime = System.currentTimeMillis();;

        return endTime - startTime;
    }

    public void zero() {
        endTime = System.currentTimeMillis();
        startTime = System.currentTimeMillis();
    }

    public String toString() {
        if (isBusy) {
            stop();
        }

        return "[" + name + "] Time elapsed: " + (getElaspedTimeInMs()) + "ms";
    }

}

File: app/src/main/java/com/lbynet/phokus/utils/Timer.kt
package com.lbynet.phokus.utils

import kotlin.jvm.JvmOverloads

class Timer @JvmOverloads constructor(private val name: String = "DEFAULT TIMER") {

    private var startTime: Long = 0
    private var endTime: Long = 0
    var isBusy = false
        private set

    fun start() {
        startTime = System.currentTimeMillis()
        isBusy = true
    }

    fun stop() {
        endTime = System.currentTimeMillis()
        isBusy = false
    }

    val elaspedTimeInMs: Long
        get() {
            if (isBusy) endTime = System.currentTimeMillis()
            return endTime - startTime
        }

    fun zero() {
        endTime = System.currentTimeMillis()
        startTime = System.currentTimeMillis()
    }

    override fun toString(): String {
        if (isBusy) {
            stop()
        }
        return "[" + name + "] Time elapsed: " + elaspedTimeInMs + "ms"
    }

    init {
        startTime = System.currentTimeMillis()
        isBusy = true
    }
}